From 652e8a3f0ddff5a0c11ad0fb6d17741077f2b011 Mon Sep 17 00:00:00 2001
From: Himanshu Madhani <himanshu.madhani@qlogic.com>
Date: Tue, 16 Apr 2013 18:11:10 +0000
Subject: [PATCH 549/858] qlcnic: fix beaconing test for 82xx adapter

o Commit 319ecf121e1da3d75dd1bde32fed255532e61797
  ("qlcnic: 83xx sysfs routines") introduced regression
  for beaconing test while refactoring 82xx code. This patch is to
  revert code to fix beaconing test for 82xx adapter.

Signed-off-by: Himanshu Madhani <himanshu.madhani@qlogic.com>
Signed-off-by: Shahed Shaikh <shahed.shaikh@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 361cd29cf9363505c2a35bbf9a034a22feebfb07)

Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
---
 drivers/net/ethernet/qlogic/qlcnic/qlcnic_sysfs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sysfs.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sysfs.c
index 987fb6f..5ef328a 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sysfs.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sysfs.c
@@ -200,10 +200,10 @@ beacon_err:
  }
 
  err = qlcnic_config_led(adapter, b_state, b_rate);
- if (!err)
+ if (!err) {
   err = len;
- else
   ahw->beacon_state = b_state;
+ }
 
  if (test_and_clear_bit(__QLCNIC_DIAG_RES_ALLOC, &adapter->state))
   qlcnic_diag_free_res(adapter->netdev, max_sds_rings);
-- 
1.8.3.1